model/host/db.go

```diff
@@ -460,6 +460,26 @@ func FindByShouldConvertProvisioning() ([]Host, error) {
 		HasContainersKey:    bson.M{"$ne": true},
 		ParentIDKey:         bson.M{"$exists": false},
 		NeedsReprovisionKey: bson.M{"$in": []ReprovisionType{ReprovisionToNew, ReprovisionToLegacy}},
+		"$or": []bson.M{
+			{NeedsNewAgentKey: true},
+			{NeedsNewAgentMonitorKey: true},
+		},
+	}))
+}
+
+// FindByNeedsJasperRestart finds all hosts that are ready and waiting to
+// restart their Jasper service.
+func FindByNeedsJasperRestart() ([]Host, error) {
+	return Find(db.Query(bson.M{
+		StatusKey:           bson.M{"$in": []string{evergreen.HostProvisioning, evergreen.HostRunning}},
+		RunningTaskKey:      bson.M{"$exists": false},
+		HasContainersKey:    bson.M{"$ne": true},
+		ParentIDKey:         bson.M{"$exists": false},
+		NeedsReprovisionKey: ReprovisionJasperRestart,
+		"$or": []bson.M{
+			{StartedByKey: bson.M{"$ne": evergreen.User}},
```

Comment:
This means that we plan to restart jasper on spawn hosts. Is that the intent?

Reply:
Yes, Jasper is designed to be restarted on all hosts. This only matters if the spawn host user happens to be running some script through Jasper and Jasper gets restarted at that exact second (which is something Jonathan wrote I think), but I honestly don't think it's worth handling.